### DISCUSSION TITLE
Query: Quantified advantage of using System.Text.Json over Newtonsoft

### DISCUSSION THREAD

**Discussion by snjosephms** [dotnet/runtime#80070] [Q&A] [answered]
The official documentation mentions that "System.Text.Json focuses primarily on performance, security, and standards compliance." Can you help quantify the advantage System.Text.Json offers in these aspects over Newtonsoft.Json?

Reference: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?pivots=dotnet-6-0#table-of-differences-between-newtonsoftjson-and-systemtextjson

**Comment by ghost** [comment]
Tagging subscribers to this area: @dotnet/area-system-text-json, @gregsdennis
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.
<details>
<summary>Issue Details</summary>
<hr />

The official documentation mentions that "System.Text.Json focuses primarily on performance, security, and standards compliance." Can you help quantify the advantage System.Text.Json offers in these aspects over Newtonsoft.Json?

Reference: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/migrate-from-newtonsoft?pivots=dotnet-6-0#table-of-differences-between-newtonsoftjson-and-systemtextjson

<table>
  <tr>
    <th align="left">Author:</th>
    <td>snjosephms</td>
  </tr>
  <tr>
    <th align="left">Assignees:</th>
    <td>-</td>
  </tr>
  <tr>
    <th align="left">Labels:</th>
    <td>

`area-System.Text.Json`, `untriaged`

</td>
  </tr>
  <tr>
    <th align="left">Milestone:</th>
    <td>-</td>
  </tr>
</table>
</details>

**Comment by snjosephms** [comment]
Following up on this, can you help with this query?

**Comment by snjosephms** [comment]
Following up on this again, can you help with this query?

**Reply by gregsdennis** [reply]
@snjosephms please have some patience. Your question was asked over the Christmas/New-Year/Other-Holiday break, and the people who can answer may not yet be

- back in the office
- caught up on other stacked up work

They're pretty good about answering questions around here.

**Reply by snjosephms** [reply]
Sure, thanks.

**Comment by eiriktsarpalis** [comment, marked as answer]
Hi @snjosephms. I'm guessing this is primarily a question about performance since the "security" and "standards compliance" bits are expanded on in the sections following the sentence that you quoted. 

It's a good question, we run a number of benchmarks comparing STJ to other well-known JSON serializers. We generally avoid posting performance numbers in documentation since these are subject to change, but here is a snapshot containing results from a few months back: https://github.com/dotnet/performance/pull/2740

**Reply by snjosephms** [reply]
Thank you, do you also have the performance comparison of STJ with the Newtonsoft library? If so, can you share the numbers for the same?

**Reply by eiriktsarpalis** [reply]
It's there, see the "JSON.NET" rows in the linked benchmark results.
